File: Python/lark_card_kit/models/elements/interactive_container.py
"""InteractiveContainer element for interactive wrappers."""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any

from .base import Element


@dataclass
class InteractiveContainer(Element):
    """Interactive container element for making content interactive.

    Wraps other elements to add interactive behaviors.
    """

    tag: str = field(default="interactive_container", init=False)
    elements: list[Any] = field(default_factory=list)
    behaviors: list[Any] | None = None
    padding: str | None = None
    margin: str | None = None
    background_style: str | None = None

File: TypeScript/LarkCardKit/src/models/elements/InteractiveContainer.ts
import type { ElementBase, Element } from './Element';

export interface InteractiveContainer extends ElementBase {
  tag: 'interactive_container';
  elements: Element[];
  actions?: InteractiveAction[];
}

export interface InteractiveAction {
  tag: string;
  element_id?: string;
  value?: Record<string, unknown>;
}
